packages/iris-grid/src/IrisGridTreeTableModel.ts

```diff
@@ -203,19 +254,32 @@ class IrisGridTreeTableModel extends IrisGridTableModelTemplate<
   }
 
   get columns(): DhType.Column[] {
-    return this.getCachedColumns(this.virtualColumns, super.columns);
+    return this.getCachedColumns(
+      this.formatter.showExtraGroupColumn ? this.virtualColumns : [],
```

Comment:
You shouldn't need to check `this.formatter.showExtraGroupColumn` anywhere else except when setting `virtualColumns`.
All these methods are already checking `virtualColumns` and behaving accordingly if there are or are not virtual columns.

Also, you're using a new empty array each time this method is being called, which is bad for memoization (i.e. it will never be memoized, so you're going to get a new column set every time).